Scripts/RemediationScripts/Remediate-EnableEncryptionInTransitForStorageAccounts.ps1

```diff
@@ -323,147 +413,269 @@ function Enable-StorageEncryptionInTransit
+    $stgWithEnableHTTPS = @()
+    #Storage Account with disabled Https only
+    $stgWithDisableHTTPS = @()
+
+    $storageAccounts | ForEach-Object {
+        try 
+        {
+            Write-Host "Fetching secure tranfer configuration of Storage Account: [$($_.StorageAccountName)]..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
 
-    try
-    {
-        #Storage Account with enabled Https only
-        $stgWithEnableHTTPS = @()
-        #Storage Account with disabled Https only
-        $stgWithDisableHTTPS = @()
-       
-        Write-Host "[step 3 of 4] : Getting the Count of Storage Account in a subscription which are having disabled 'secure transfer'."
-        $storageAccounts | ForEach-Object {
             if ($_.EnableHttpsTrafficOnly)
             {
                 $stgWithEnableHTTPS += $_ 
+                $logResource = @{}
+                $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+                $logResource.Add("ResourceName",($_.ResourceName))
+                $logResource.Add("Reason","Secure Transfer is already enabled.")    
+                $logSkippedResources += $logResource
             }        
             else
             { 
-                 $stgWithDisableHTTPS += $_|Select-Object  @{N='ResourceId';E={$_.Id}},
-                                                           @{N='ResourceGroupName';E={$_.ResourceGroupName}},
-                                                           @{N='StorageAccountName';E={$_.StorageAccountName}},
-                                                           @{N='EnableHttpsTrafficOnly';E={$_.EnableHttpsTrafficOnly}}                                                                  
+                $stgWithDisableHTTPS += $_|Select-Object  @{N='ResourceId';E={$_.Id}},
+                                                        @{N='ResourceGroupName';E={$_.ResourceGroupName}},
+                                                        @{N='StorageAccountName';E={$_.StorageAccountName}},
+                                                        @{N='EnableHttpsTrafficOnly';E={$_.EnableHttpsTrafficOnly}}                                                                  
             }
-        }                    
-                              
-        $totalStgWithEnableHTTPS = ($stgWithEnableHTTPS | Measure-Object).Count
-        $totalStgWithDisableHTTPS = ($stgWithDisableHTTPS | Measure-Object).Count
-    
-        Write-Host "Storage Account(s) with enabled 'secure transfer': [$($totalStgWithEnableHTTPS)]" -ForegroundColor $([Constants]::MessageType.Update)
-        Write-Host "Storage Account(s) with disabled 'secure transfer': [$($totalStgWithDisableHTTPS)]" -ForegroundColor $([Constants]::MessageType.Update)
+            Write-Host "Successfully fetched secure tranfer configuration of Storage Account: [$($_.StorageAccountName)]" -ForegroundColor $([Constants]::MessageType.Update)
```

Comment:
Here we are not making any API call, do we still need to print above line for each storage. Just summary in the end of this steps would be better.

Reply:
Summary is also one of the way but this is how it is done in most of the similar type BRS. So to have consistency as much as possible. I did it this way.